starboard/extension/system_info.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_EXTENSION_SYSTEM_INFO_H_
+#define STARBOARD_EXTENSION_SYSTEM_INFO_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define kStarboardExtensionSystemInfoName "dev.starboard.extension.SystemInfo"
+
+typedef struct StarboardExtensionSystemInfoApi {
+  // Name should be the string |kStarboardExtensionSystemInfoName|.
+  // This helps to validate that the extension API is correct.
+  const char* name;
+
+  // This specifies the version of the API that is implemented.
+  uint32_t version;
+
+  // The fields below this point were added in version 1 or later.
+
+  int64_t (*GetAppStartWithAndroidFix)();
```

Comment:
More of a question, not a comment. Do you want to leave it as a very specific API for retrieving this one number ? Or, make it a  bit more generic with 

```
int64_t (*GetSystemMetric)(const char *name)
```
where `name="appstart_with_android_fix"` ? With the latter, we wouldn't need to update the interface  extension version every time someone wants to retrieve a value.

There's as strong YAGNI argument for the latter here, but i also know Media team is trying to do similar things.

Reply:
I'd rather leave it very specific with the goal of removing it in a later minor release if possible—it's not a particularly hard thing to implement so I'm not that worried about it potentially being redone

Reply:
Sgtm.